Apps/WebClient/src/ClientApp/src/components/modal/addQuickLink.vue

```diff
@@ -0,0 +1,321 @@
+                this.populateEnabledQuickLinks();
+                const existingQuickLinks = this.quickLinks;
+                return this.getAvailabeQuickLinks(
+                    existingQuickLinks,
+                    this.enabledQuickLinks
+                );
+            }
+            this.populateEnabledQuickLinks();
+            return this.enabledQuickLinks;
+        }
+    }
+
+    private get isAddButtonDisabled(): boolean {
+        return (
+            this.enabledQuickLinkFilter.length == 0 ||
+            (this.enabledQuickLinkFilter.length > 0 &&
+                this.selectedQuickLinks.length == 0)
+        );
+    }
+
+    private get isCheckBoxChecked(): boolean {
+        return this.enabledQuickLinkFilter.length == 0;
+    }
+
+    private addQuickLink(submittedQuickLinks: QuickLink[]): Promise<void> {
+        const quickLinks: QuickLink[] = (this.quickLinks ?? []).concat(
+            submittedQuickLinks
+        );
+        return this.updateQuickLinks({
+            hdid: this.user.hdid,
+            quickLinks: quickLinks,
+        });
+    }
+
+    private created() {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    }
+
+    private getAvailabeQuickLinks(
+        existingQuickLinks: QuickLink[],
+        enabledQuickLinks: QuickLinkFilter[]
+    ): QuickLinkFilter[] {
+        existingQuickLinks.forEach((element) =>
+            this.logger.debug(
+                `Available Quick Links - Existing store quick links:  ${element}`
+            )
+        );
+
+        enabledQuickLinks.forEach((element) =>
+            this.logger.debug(
+                `Available Quick Links - Enabled quick links:  ${element}`
+            )
+        );
+
+        let result: QuickLinkFilter[] = [];
+        result = enabledQuickLinks.filter((enabledQL) => {
+            return !existingQuickLinks.find((existingQL) => {
+                return existingQL.name === enabledQL.name;
+            });
+        });
+        return result;
+    }
+
+    private getCheckedValue(quickLink: QuickLinkFilter): string {
+        return (
+            '{ "name": "' +
+            quickLink.name +
+            '", "filter": { "modules": [ "' +
+            quickLink.module +
+            '"] }}'
+        );
+    }
+
+    private populateEnabledQuickLinks(): void {
+        this.enabledQuickLinks = this.quickLinkFilter.filter(
+            (filter: QuickLinkFilter) => filter.isEnabled
+        );
+        this.logger.debug(`Enabled quick links:  ${this.enabledQuickLinks}`);
+    }
+
+    private forceCheckboxComponentRerender(): void {
+        this.logger.debug(
+            `Checkbox Component Key before force re-render: ${this.checkboxComponentKey}`
+        );
+        this.checkboxComponentKey++;
+        this.logger.debug(
+            `Checkbox Component Key after force re-render: ${this.checkboxComponentKey}`
+        );
+    }
+
+    private removeSelectedQuickLink(quickLinkName: string) {
```

Comment:
This shouldn't be needed now that the bug relating to updating user preferences has been fixed.

Reply:
Removed as store getter bug has been fixed.